lib/rubocop/cop/style/struct_inheritance.rb

```diff
@@ -0,0 +1,42 @@
+# encoding: utf-8
+
+module RuboCop
+  module Cop
+    module Style
+      # This cop checks for inheritance from Struct.new.
+      #
+      # @example
+      #   # bad
+      #   class Person < Struct.new(:first_name, :last_name)
+      #   end
+      #
+      #   # good
+      #   Person = Struct.new(:first_name, :last_name)
+      class StructInheritance < Cop
+        MSG = "Don't extend an instance initialized by Struct.new. Extending " \
```

Comment:
One final remark. Keep just the first sentence of the MSG (long messages look bad with some formatters) and wrap `Struct.new` in backquotes.